### PR TITLE
Empty todo bug fix and new add feature

### PR DESCRIPTION
**Pull request by philipAthanasopoulos**
Case where user could input an empty TODO task fixed on frontend.
User can now create new todo by pressing ENTER key

image

### CODE REVIEW

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 1]
```diff
@@ -9,21 +9,29 @@ export const TodoInputField = () => {
...
   const addTodoEntity = useCallback(() => {
-    const todoEntity: ITodoEntity = {
-      name: input,
-      done: false,
-      createdAt: new Date().toISOString(),
-      updatedAt: new Date().toISOString(),
-    };
-    dispatch(insertTodoEntity(todoEntity));
-    setInput("");
+    if (input.trim() == "") alert("TODO field cannot be empty")
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: in general it is better to use === in TypeScript. Which will also check for the type of the variable.
In this case `if (!!input.trim())` will also do the trick

Comment (Kitsopappas) [reply web-app/uni-todo-app/src/components/TodoInputField.tsx]: An early return is also more beneficial and readable.

```
if (!!input.trim()) {
// I would also replace the alert with something else (https://v1.tailwindcss.com/components/alerts)
 return;
}
const todoEntity: ITodoEntity = {
        name: input,
        done: false,
        createdAt: new Date().toISOString(),
        updatedAt: new Date().toISOString(),
};
dispatch(insertTodoEntity(todoEntity));
setInput("");

```

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 2]
```diff
@@ -9,21 +9,29 @@ export const TodoInputField = () => {
...
+      setInput("");
+    }
   }, [input, dispatch]);
 
   return (
     <div className="flex mt-4">
       <input
         value={input}
         onChange={(e) => setInput(e.target.value)}
+        onKeyUp={(e) => {
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: maybe `onKeyPress` is better ;)

File: backend/UniTodo/.editorconfig [hunk 3]
```diff
@@ -2,3 +2,85 @@
 
 # IDE0058: Expression value is never used
 csharp_style_unused_value_expression_statement_preference = unused_local_variable
+csharp_using_directive_placement = outside_namespace:silent
```
Comment (Kitsopappas) [inline backend/UniTodo/.editorconfig]: can you have this file on a separate commit with a description on why these changes are needed?

**Review by Kitsopappas — CHANGES_REQUESTED** [review]

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 4]
```diff
@@ -9,14 +9,17 @@ export const TodoInputField = () => {
...
-      name: input,
-      done: false,
-      createdAt: new Date().toISOString(),
-      updatedAt: new Date().toISOString(),
-    };
-    dispatch(insertTodoEntity(todoEntity));
-    setInput("");
+    if (input.trim() == "") alert("TODO field cannot be empty")
+    else {
+      const todoEntity: ITodoEntity = {
```
Comment (GiotisK) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: I  would assign the date to a new constant and reuse it in both `createdAt` , `updatedAt`

File: backend/UniTodo/.editorconfig (hunk 3 above)
Comment (philipAthanasopoulos) [reply backend/UniTodo/.editorconfig]: I found out that this is a file Visual Studio adds to enforce specific styling on the editor. So perhaps this may not break anything, but it also doesn't need to be merged? I could revert this change

Comment (Kitsopappas) [reply backend/UniTodo/.editorconfig]: Feel free to leave it here. Just add it on a separate commit ;) So we have a cleaner history

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 5]
```diff
@@ -5,34 +5,62 @@ import { useAppDispatch } from "../hooks";
...
-      createdAt: new Date().toISOString(),
-      updatedAt: new Date().toISOString(),
-    };
-    dispatch(insertTodoEntity(todoEntity));
-    setInput("");
+    if (input.trim() === "") {
+      setError(" TODOs cannot be empty!");
+    } 
+    
+    else {
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: No need for the else here! Early return.
Also don't forget about `if (!!input.trim())`

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 6]
```diff
@@ -5,34 +5,62 @@ import { useAppDispatch } from "../hooks";
...
-    const todoEntity: ITodoEntity = {
-      name: input,
-      done: false,
-      createdAt: new Date().toISOString(),
-      updatedAt: new Date().toISOString(),
-    };
-    dispatch(insertTodoEntity(todoEntity));
-    setInput("");
+    if (input.trim() === "") {
+      setError(" TODOs cannot be empty!");
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: Leading space not needed

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 7]
```diff
@@ -5,34 +5,62 @@ import { useAppDispatch } from "../hooks";
...
-        placeholder="Add Todo"
-      ></input>
-      <button
-        onClick={addTodoEntity}
-        className="flex-no-shrink p-2 border-2 rounded text-teal border-teal hover:text-white hover:bg-teal"
-      >
-        Add
-      </button>
+    <div>
+        {error && (
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: Better add the new error case on a separate component and use it here instead ;)

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 8]
```diff
@@ -5,34 +5,62 @@ import { useAppDispatch } from "../hooks";
...
+          <div className="bg-teal-100 border-t-4 border-teal-500 rounded-b text-teal-900 px-4 py-3 shadow-md" role="alert">
+            <div className="flex">
+              <div className="py-1"><svg className="fill-current h-6 w-6 text-teal-500 mr-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M2.93 17.07A10 10 0 1 1 17.07 2.93 10 10 0 0 1 2.93 17.07zm12.73-1.41A8 8 0 1 0 4.34 4.34a8 8 0 0 0 11.32 11.32zM9 11V9h2v6H9v-4zm0-6h2v2H9V5z"/></svg></div>
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx]: Add the SVG into a file and include it like a component

**Review by Kitsopappas — CHANGES_REQUESTED** [review]

File: web-app/uni-todo-app/src/components/TodoInputField.tsx [hunk 9]
```diff
@@ -2,13 +2,20 @@ import { useCallback, useState } from "react";
...
+import { AlertWindow } from "./AlertWindow";
 
 export const TodoInputField = () => {
   const [input, setInput] = useState("");
+  const [error, setError] = useState("");
 
   const dispatch = useAppDispatch();
 
   const addTodoEntity = useCallback(() => {
+    if(!input.trim()) {
```
Comment (Kitsopappas) [inline web-app/uni-todo-app/src/components/TodoInputField.tsx:14]: You want to show the error only when the input value is empty. So better use `!!input.trim()`

Check this out https://webionista.com/javascript-double-exclamation-mark/#:~:text=In%20short%2C%20The%20double%20exclamation,for%20use%20in%20conditional%20statements.

**Review by Kitsopappas — CHANGES_REQUESTED** [review]

**Comment by Kitsopappas** [comment]
Please squash and make your commits more descriptive. After the last change I requested you should be able to merge ;)

File: web-app/uni-todo-app/src/components/TodoInputField.tsx (hunk 9 above)
Comment (philipAthanasopoulos) [reply web-app/uni-todo-app/src/components/TodoInputField.tsx:14]: Yeah I'm a bit confused on that. When i use !! instead, an empty value is accepted. If the string is empty then from what i understand !!input.trim() is false, so if the condition is false the block inside will not run

Comment (Kitsopappas) [reply web-app/uni-todo-app/src/components/TodoInputField.tsx:14]: My bad! Long day :P

**Review by Kitsopappas — APPROVED** [review]

**Review by GiotisK — APPROVED** [review]
